src/controllers/buildingsController.js

```diff
@@ -79,6 +123,12 @@ const updateBuilding = async (req, res) => {
 
 const deleteBuilding = async (req, res) => {
   try {
+    const appointment = await models.Appointments.find({ building: req.params.id });
```

Comment:
What happens if you have a building but it's not used by an appointment? The find method returns an empty array and, in the next line when you do `if (appointment)` this will be true because you have an empty array and it's not null.
So, the best way to do this is using findOne instead of find. findOne will return the first appointment that have a building with the provided id. If there is not an appointment with that building id, the result will be null.

```suggestion
    const appointment = await models.Appointments.findOne({ building: req.params.id });
```